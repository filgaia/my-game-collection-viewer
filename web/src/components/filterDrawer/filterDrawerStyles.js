const filterDrawerStyles = (theme) => ({
    listItem: {
        marginRight: 0,
        width: '100%'
    }
});

export default filterDrawerStyles;
