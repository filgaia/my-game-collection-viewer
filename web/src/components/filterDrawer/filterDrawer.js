/* eslint-disable react/prop-types */
// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// @material
import {
  Drawer,
  IconButton,
  List,
  ListItem,
  ListItemIcon,
  Tooltip,
} from '@mui/material';
// https://material.io/tools/icons/?style=baseline
import {
  FilterList,
} from '@mui/icons-material';
import withStyles from '@mui/styles/withStyles';
// @componets
import LabelTag from '../labelTag/labelTag';
// @styles
import filterDrawerStyles from './filterDrawerStyles';
// @constants
import { FILTER_DRAWER } from '../../constants/index';

export class FilterDrawer extends Component {
  constructor(props) {
    super(props);

    this.onToggleDrawer = this.onToggleDrawer.bind(this);
  }

  onToggleDrawer(open) {
    const { toggleDrawer } = this.props;
    const response = { drawer: FILTER_DRAWER, open };
    toggleDrawer({ response });
  }

  buildItems() {
    const { classes, gamesInformation, setLabelFilter } = this.props;
    const labelsTagsItems = gamesInformation.get('labels');

    const Tag = React.forwardRef(({ item }, ref) => (
      <div className={classes.tagDiv} ref={ref}>
        <LabelTag
          key={item.id}
          isDrawer
          label={item}
          gamesInformation={gamesInformation}
          setLabelFilter={setLabelFilter}
        />
      </div>
    ));

    const items = labelsTagsItems ? labelsTagsItems.map((item, index) => (
      <ListItem button key={`filter-${index}`}>
        <ListItemIcon className={classes.listItem}>
          <Tooltip title={item.name} aria-label={item.name}>
            <Tag item={item} />
          </Tooltip>
        </ListItemIcon>
      </ListItem>
    )) : null;

    return (
      <List>
        {items}
      </List>
    );
  }

  render() {
    const { filterDrawer } = this.props;
    const items = this.buildItems();

    return (
      <>
        <IconButton color="inherit" onClick={() => this.onToggleDrawer(true)} size="large">
          <Tooltip title="Filter..." aria-label="Filter...">
            <FilterList />
          </Tooltip>
        </IconButton>
        <Drawer anchor="right" open={filterDrawer} onClose={() => this.onToggleDrawer(false)}>
          <div
            tabIndex={0}
            role="button"
            onClick={() => this.onToggleDrawer(false)}
            onKeyDown={() => this.onToggleDrawer(false)}
          >
            {items}
          </div>
        </Drawer>
      </>
    );
  }
}

FilterDrawer.propTypes = {
  classes: PropTypes.object.isRequired,
  filterDrawer: PropTypes.bool.isRequired,
  gamesInformation: PropTypes.object.isRequired,
  setLabelFilter: PropTypes.func.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
};

export default withStyles(filterDrawerStyles)(FilterDrawer);
