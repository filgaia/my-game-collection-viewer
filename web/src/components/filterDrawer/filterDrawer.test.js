import React from 'react';
import renderer from 'react-test-renderer';
import { FilterDrawer } from './filterDrawer';

describe('<FilterDrawer />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FilterDrawer
        classes={{}}
        filterDrawer={false}
        gamesInformation={{ get: jest.fn(() => []) }}
        setLabelFilter={jest.fn()}
        toggleDrawer={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
