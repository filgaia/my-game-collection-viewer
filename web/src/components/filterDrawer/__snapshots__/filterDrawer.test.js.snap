// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterDrawer /> renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <svg
    aria-hidden={true}
    aria-label="Filter..."
    aria-labelledby={null}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-mui-internal-clone-element={true}
    data-testid="FilterListIcon"
    focusable="false"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
    />
  </svg>
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
